core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2125,6 +2125,15 @@ package object config {
       .intConf
       .createWithDefault(1)
 
+  private[spark] val IO_COMPRESSION_ZSTD_STRATEGY =
+    ConfigBuilder("spark.io.compression.zstd.strategy")
+      .doc("Compression strategy for Zstd compression codec. The higher the value is, the more " +
+        "complex it becomes, usually resulting stronger but slower compression or higher CPU " +
+        "cost.")
+      .version("4.1.0")
+      .intConf
```

Comment:
Currently, the zstd version we are using supports only levels 0 to 9. I chose to skip validation for the following reasons: 1) the range of values may expand in the future, and 2) it is generally difficult to misuse since users are likely familiar with this feature of zstd before using it.